File: ios/Vegviser/UnityBridge.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(UnityMessageReceiver, RCTEventEmitter)

RCT_EXTERN_METHOD(onUnityMessage:(NSString *)message)

@end

File: ios/Vegviser/UnityBridge.swift
import Foundation
import React

@objc(UnityMessageReceiver)
class UnityMessageReceiver: RCTEventEmitter {
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func supportedEvents() -> [String]! {
        return ["UnityMessage"]
    }
    
    @objc
    func onUnityMessage(_ message: String) {
        sendEvent(withName: "UnityMessage", body: ["message": message])
    }
}

